sycl/test/check_device_code/device_has.cpp

```diff
@@ -7,30 +7,30 @@
 using namespace sycl;
 queue q;
 
-// CHECK: define weak_odr dso_local spir_kernel void @{{.*}}kernel_name_1{{.*}} !sycl_declared_aspects ![[ASPECTS1:[0-9]+]] !srcloc ![[SRCLOC1:[0-9]+]] {{.*}}
```

Comment:
Are we guaranteed any order with `// CHECK-SAME: !sycl_used_aspects ![[ASPECTS1]]` below or is FileCheck smart enough to check the entire line again when `CHECK-SAME` is used? If so, could we maybe make the `!srcloc` check a `CHECK-SAME` line too?

Reply:
Hi, @steffenlarsen 
I tried this approach like following:

// CHECK: define weak_odr dso_local spir_kernel void @{{.*}}kernel_name_1{{.*}} !sycl_declared_aspects ![[ASPECTS1:[0-9]+]]
// CHECK-SAME:!srcloc ![[SRCLOC1:[0-9]+]] {{.*}}

However, FileCheck is not smart enough, the CHECK-SAME will fail to report that "is not on the same line as previous match". 
I also tried following:
// CHECK: define weak_odr dso_local spir_kernel void @{{.*}}kernel_name_1{{.*}}
// CHECK-SAME-DAG: !srcloc ![[SRCLOC1:[0-9]+]] {{.*}}
// CHECK-SAME-DAG: !sycl_declared_aspects ![[ASPECTS1:[0-9]+]]

However, FileCheck will report "undefined variable: ASPECTS1" for lines using ASPECTS1.
Do you have any idea to address these issues?
Thanks very much.

Reply:
Does 

```c++
// CHECK: define weak_odr dso_local spir_kernel void @{{.}}kernel_name_1{{.}} !sycl_declared_aspects ![[ASPECTS1:[0-9]+]]
// CHECK-SAME-DAG: !srcloc ![[SRCLOC1:[0-9]+]]
```

work? It should be fine to keep the `!sycl_declared_aspects` in the same line.

Alternatively, we could run FileCheck twice and have

```c++
// CHECK-DECL-ASPECTS: define weak_odr dso_local spir_kernel void @{{.}}kernel_name_1{{.}} !sycl_declared_aspects ![[ASPECTS1:[0-9]+]]
// CHECK-SRCLOC: define weak_odr dso_local spir_kernel void @{{.}}kernel_name_1{{.}} !srcloc ![[SRCLOC1:[0-9]+]]
```

but with that we could also just split it into separate tests.

Reply:
Hi, @steffenlarsen 
Move srcloc to check-same-dag can't work either, FileCheck will match srcloc in next line and report the match is not in the same line.
Run FileCheck twice for different check-prefix can work, I just updated the test, could you help review?
Thanks very much.